docs/book/src/developers/development.md

```diff
@@ -0,0 +1,220 @@
+```
+
+All development dependencies should be taken care of via Devbox and/or make target dependencies.
+
+#### Using devbox
+
+1. Install dependent packages in your project 
+   ```shell
+   devbox install
+   ```
+
+   ```admonish success title=""
+   This will take a while, go and grab a drink of water.
+   ```
+
+2. Use devbox environment
+   ```shell
+   devbox shell
+   ```
+
+From this point you can use the devbox shell like a regular shell. 
+The rest of the guide assumes a devbox shell is used, but the make target
+dependencies will install any missing dependencies if needed when running
+outside of a devbox shell.
+
+### Using tilt
+To build a kind cluster and start Tilt, simply run:
+```shell
+make tilt-cluster
+```
+
+Once your kind management cluster is up and running, you can
+[deploy a workload cluster](#deploying-a-workload-cluster).
+
+To tear down the tilt-cluster, run
+
+```shell
+kind delete cluster --name tilt
+```
+
+### Deploying a workload cluster
+
+After your kind management cluster is up and running with Tilt, you should be ready to deploy your first cluster.
+
+#### Customizing the cluster deployment
+
+Here is a list of required configuration parameters:
+
+```bash
+# Cluster settings
+export CLUSTER_NAME=capl-cluster
+export KUBERNETES_VERSION=1.29.1
+export LINODE_REGION=us-ord
+
+# Linode settings
+export LINODE_TOKEN=<your linode PAT>
+
+# Machine settings
+export LINODE_OS=linode/ubuntu22.04
+export LINODE_CONTROL_PLANE_MACHINE_TYPE=g6-standard-2
+export LINODE_MACHINE_TYPE=g6-standard-2
+export CONTROL_PLANE_MACHINE_COUNT=3
+export WORKER_MACHINE_COUNT=3
+
+# Generate SSH key.
+# If you want to provide your own key, skip this step and set LINODE_SSH_KEY to your existing public key.
+SSH_KEY_FILE=.linodeSSHkey
+rm -f "${SSH_KEY_FILE}"
+ssh-keygen -t rsa -b 4096 -f "${SSH_KEY_FILE}" -N '' 1>/dev/null
+export LINODE_SSH_KEY="$(cat "${SSH_KEY_FILE}.pub)"
+```
+
+~~~admonish tip
+You can also use `clusterctl generate` to see which variables need to be set:
+
+```
+clusterctl generate cluster $CLUSTER_NAME --from ./templates/cluster-template.yaml --list-variables
+```
+
+~~~
+
+```admonish warning
+Please note the templates require the use of `clusterctl generate` to substitute the environment variables properly.
+```
+
+#### Creating the workload cluster
+
+Once you have all the necessary environment variables set,
+you can deploy a workload cluster with the following command:
+
+```shell
+clusterctl generate cluster $CLUSTER_NAME \
+  --kubernetes-version v1.29.1 \
+  --from templates/cluster-template.yaml \
+  | kubectl apply -f -
+```
+
+This will provision the cluster with the CNI defaulted to [cilium](../topics/addons.md#cilium)
+and the [linode-ccm](../topics/addons.md#ccm) installed.
+
+Next, obtain the provisioned cluster kubeconfig:
+```bash
+clusterctl get kubeconfig $CLUSTER_NAME > $CLUSTER_NAME-kubeconfig
+```
+
+Once the cluster is reachable, create the Kubernetes Secret on the provisioned
```

Comment:
just to track the documentation side of this change as well, can we have this secret created as part of the `clusterctl generate` templates?